src/common/services/Solana/GetData/GenerateTokenOwnedList.tsx

```diff
@@ -0,0 +1,42 @@
+import { TokenInfoMap } from '@solana/spl-token-registry';
+import ReadTokenWallet from './ReadTokenWallet';
+
+import { ItokenOwned } from '../../../types';
+import defaultUserProfileImage from '../../../../assets/defaultUserProfileImage.png';
+
+/* TODO:
+Once you get the community db design 
+Fetch the community member count
+*/
+export default async function GenerateTokenOwnedList(
+  tokenRegistry: TokenInfoMap,
+  publicKey: string
+): Promise<ItokenOwned[]> {
+  const tokenOwned = await ReadTokenWallet(publicKey);
+  const tokenOwnedList = new Array<ItokenOwned>();
+
+  for (let i = 0; i < tokenOwned.length; i += 1) {
+    const tokenInfos = tokenRegistry.get(tokenOwned[i].mint);
+    if (tokenInfos !== undefined) {
+      // add to the beginning
+      tokenOwnedList.unshift({
+        mint: tokenOwned[i].mint,
+        name: tokenInfos.symbol,
+        memberCount: 0,
+        amountHeld: tokenOwned[i].uiAmount,
+        imageUrl: tokenInfos.logoURI,
+      });
+    } else {
+      // add to the end
+      tokenOwnedList.push({
+        mint: tokenOwned[i].mint,
+        name: 'Unknown',
+        memberCount: 9,
```

Comment:
what is the memberCount and why is it 9?

Reply:
Was waiting for decision on database structure

Reply:
* the todo at the top

Reply:
changed